### PR TITLE
MF-898 - Add bulk connect to CLI and SDK

### LINKED ISSUE #898: Add bulk connect
**FEATURE REQUEST**

1. Is there an open issue addressing this request? If it does, please add a "+1" reaction to the
   existing issue, otherwise proceed to step 2.
No

2. Describe the feature you are requesting, as well as the possible use case(s) for it.
As described [here](https://github.com/mainflux/mainflux/issues/484#issuecomment-544544684)
  Now when we have bulk creation of things and channels, thanks to #889  it would be nice to have an API that can connect things to channels in bulk.

3. Indicate the importance of this feature to you (must-have, should-have, nice-to-have).
Nice to have

### PR DESCRIPTION
**Pull request by nwneisen**
### What does this do?
Adds bulk connect functionality to the SDK and CLI

### Which issue(s) does this PR fix/relate to?
Resolves #898 

### List any changes that modify/break current functionality
Multiple connections can now be created using the SDK or CLI

### Have you included tests for your changes?
Yes

### Did you document any new/modified functionality?
Yes

### CODE REVIEW

File: sdk/go/things_test.go [hunk 1]
```diff
@@ -694,6 +694,111 @@ func TestConnectThing(t *testing.T) {
...
+			thingID: thingID,
+			chanID:  chanID1,
+			token:   token,
+			err:     nil,
+		},
+
+		{
+			desc:    "connect existing things to non-existing channels",
+			thingID: thingID,
+			chanID:  "9",
```
Comment (manuio) [inline sdk/go/things_test.go]: You could maybe use a const here

File: docs/cli.md [hunk 2]
```diff
@@ -155,6 +155,15 @@ mainflux-cli channels get <channel_id> <user_auth_token>
 mainflux-cli things connect <thing_id> <channel_id> <user_auth_token>
 ```
 
+#### Bulk Connect Things to Channels
+
+```bash
+mainflux-cli provision connect <file> <user_auth_token>
+```
+
+* `file` - A CSV or JSON file containing thing and channel ids
```
Comment (manuio) [inline docs/cli.md:164]: Could you add an example of this please?

**Review by manuio — CHANGES_REQUESTED** [review]

File: sdk/go/things_test.go (hunk 1 above)
Comment (nwneisen) [reply sdk/go/things_test.go]: I changed a few of these to use a const

File: docs/cli.md [hunk 3]
```diff
@@ -155,6 +155,24 @@ mainflux-cli channels get <channel_id> <user_auth_token>
...
+* `user_auth_token` - A valid user auth token for the current system
+
+An example file might be
+
+```csv
+7f3216a2-f5c8-4410-a37e-a1bbd7f48231,f5dc42c8-0bf1-4850-bb0e-ddea435e7814
+f26688e7-8bdf-4ec8-b1dc-1bfb764cfa3a,ca3365f9-2202-40a6-a9d6-75b6690df466
```
Comment (manuio) [inline docs/cli.md]: Sorry, but could you use `<thing_id>` and `<channel_id>`? And also add an example for JSON, please?

**Review by manuio — CHANGES_REQUESTED** [review]

**Review by manuio — DISMISSED** [review]
LGTM!

**Review by manuio — DISMISSED** [review]
LGTM!

**Comment by nwneisen** [comment]
I'm not sure why the coverage dropped after changing to consts.  It's not giving me much information to go off of.

**Review by drasko — DISMISSED** [review]
LGTM

**Comment by drasko** [comment]
Great contribution again @nwneisen, approved. I'll let other take a look and merge tomorrow.

File: cli/provision.go [hunk 4]
```diff
@@ -275,3 +298,47 @@ func channelsFromFile(path string) ([]mfxsdk.Channel, error) {
...
+
+	file, err := os.OpenFile(path, os.O_RDONLY, os.ModePerm)
+	if err != nil {
+		return mfxsdk.ConnectionIDs{}, err
+	}
+	defer file.Close()
+
+	connections := mfxsdk.ConnectionIDs{}
+	switch filepath.Ext(path) {
+	case ".csv":
```
Comment (nmarcetic) [inline cli/provision.go]: Maybe move this to constants or no need?

Comment (manuio) [reply cli/provision.go]: IMO no need if it only appears here

Comment (nwneisen) [reply cli/provision.go]: I believe this is used in 3 places.  I was thinking I had switched the others to constants previously.  I will switch this and the json extension over

**Review by drasko — APPROVED** [review]
LGTM

**Review by anovakovic01 — APPROVED** [review]
LGTM!

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**